crates/cast/bin/cmd/call.rs

```diff
@@ -149,6 +151,8 @@ impl CallArgs {
 
                     let evm_opts = figment.extract::<EvmOpts>()?;
 
+                    config.fork_block_number = derive_fork_block_number(&provider, block).await?;
```

Comment:
I'm not exactly sure about this, 
I believe this should be set to evm_opts instead.

sorry this entire command is a bit messy...

Reply:
if it's overriden with the config block number then I think it's fine for now